framework/sandstone.cpp

```diff
@@ -1784,8 +1784,20 @@ static TestResult child_run(/*nonconst*/ struct test *test, int child_number)
         if (sApp->shmem->use_strict_runtime && wallclock_deadline_has_expired(sApp->endtime)){
             // skip cleanup on the last test when using strict runtime
         } else {
-            if (test->test_cleanup)
-                test->test_cleanup(test);
+            if (test->test_cleanup) {
+                ret = test->test_cleanup(test);
+                if (state == TestResult::Passed) {
+                    if (ret == EXIT_SKIP) {
+                        log_skip(RuntimeSkipCategory, "Skip requested in cleanup");
+                        state = TestResult::Skipped;
```

Comment:
Does this make sense? If at least one thread ran, we didn't skip...

Reply:
It could be usable in producer-consumer type tests or with tests which uses accelerators (with some "cpu processing" threads), as overall test is skipped only when all threads in run() return SKIP. Detecting that the test has "done nothing" might be simplest in cleanup(), custom logic seems to be easiest one.

Reply:
@jposwiata , if i understood this right, if a test skips on only one 1 cpu in the run function, the test is still considered a `pass`. Now if the same test returns `EXIT_SKIP` in cleanup, does this not change the result of the test to be `result:skip` instead of `result:pass` in the original implementation?

Reply:
I think it does, but that may be the intention. The `run` functions may not be able to tell that anything useful was generated, only the `cleanup` would. Hopefully, those `run` threads didn't do much work in the first place, otherwise we just wasted CPU time.

Reply:
Exactly, this is the intention. SKIP is very tricky case, I think it is (might be) used for statistical reasons (to not increase number of passes).
Speaking about "wasted time".. Identical situation is when memory allocation (with mmap) reaches limits: all threads should return -ENOMEM ("unexpected OS error"), so test will SKIP, but (potentially) a lot of work is done. (BTW, I wonder why malloc() causes test to fail, while mmap() is expected to skip..)